### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by erkshen**
Hi all,

This is a solution to Report Task 6, an implementation of the Visual Transformer model, used to classify the images from the ADNI dataset as either exhibiting Alzheimer's disease or not. This version achieves roughly 65% accuracy on the test set. The files in this pull request are:
- `README.md`, an overview of the project
- `ViT.ipynb`, a notebook version of the project used to run and test functionality in Google Colab
- `dataset.py`, contains the function to load images from zip files
- `modules.py`, contains the various components of the ViT model, such as the transformer encoder and the MLP
- `predict.py`, contains a function for classifying an image passed to it
- `train.py`, contains the functions for training the model and reporting accuracy measures
- `images/*.jpeg` and `images/*.png`, various images used for the README or for other interest.

Thank you, looking forward to any feedback!
Eric Shen, 46466369

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
Model design: Great
Data split: Great
Model training: Great
Plots: Great
README: clear explanation

Question: "The current best model trained reached 63% accuracy on the test set, with 75% testing accuracy and 71% validation accuracy" Do you mean test ACC is 63% and training ACC is 75% instead of testing ACC? It's okay test ACC is not over 80%

**Comment by erkshen** [comment]
> Do you mean test ACC is 63% and training ACC is 75% instead of testing ACC?

Hi Linfeng,
Yes, that was a typo. It should say 75% training accuracy instead of testing accuracy for the second stat there. Fixed in the most recent commit.
Thanks!

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem (slightly low performance) -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, could have more info -1
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
